### DISCUSSION TITLE
Discussion of Which Devices Should Be Supported by Scratch Addons

### DISCUSSION THREAD

**Discussion by BroJac5246** [ScratchAddons/ScratchAddons#7027] [Discussion]
In #5160, steps were laid out to support touchscreen devices. However, these devices were touchscreen laptops and Chromebook hybrids/tablets, which are very closely related to the normal devices that, until recently, have been the focus of Scratch Addons.

A few months ago, Mozilla announced that Firefox for Android would support add-ons (extensions), and support was recently added (#6789). Also recently, GarboMuffin pointed out that they would be willing to publish Scratch Addons under the same Apple Developer account used for TurboWarp (https://github.com/ScratchAddons/ScratchAddons/discussions/3452#discussioncomment-7941734). This has long been considered the largest roadblock to Scratch Addons support on Macs, but iOS/iPadOS also support extensions, and this opens the gate to Scratch Addons on those platforms as well.

A year and a half ago in #4562, Scratch Addons for mobile was rejected, but this stance widely appears to have changed since then, including multiple mobile-specific issues being filed and the Docs potentially being updated to suggest using Firefox for Android (https://github.com/ScratchAddons/website-v2/pull/405).

I think many people support Scratch Addons for other platforms, but not all of them make sense. For example, a phone probably isn't a practical device on which to use Scratch, but an iPad is. So I want to bring up the question, which of these should be supported? (I've checked browsers that we currently support):
**Desktop**
- [x] Chrome (and Chromium)
- [x] Firefox
- [ ] Safari (Mac only)

**Android**
> [!NOTE] 
> We can't distinguish between phones and tablets. Scratch Addons can already be officially installed on Firefox and Kiwi and somewhat-officially on Edge Canary using developer options.
- [ ] Firefox
- [ ] Kiwi
- [ ] Edge (in the future)

**iPadOS**
- [ ] Safari

**iOS**
- [ ] Safari

**Misc**
- [ ] General touchscreen devices via TurboWarp (https://github.com/ScratchAddons/ScratchAddons/issues/5160#issuecomment-1269049819)

I suggest we support Safari on Macs and iPads and Android on tablets, but what do you all think? iPads are a particularly tough one as they're widespread in Education but I believe supporting them requires an app, which we'd need to look into. The app could maybe be standalone or even theoretically bundled with TurboWarp, but that's complicated.

Whichever we choose, it'll take effort to port, and, potentially, maintain Scratch Addons for other devices. However, the amount of effort varies by platform, since the regular release could probably be lightly modified to work well on Android while it might need to be heavily changed to work on Safari.

**Comment by elyk233** [comment]
On safari I have tried to use the extension converter and it worked on macOS fairly well. I couldn’t port it to iPadOS, it kept timing out.

**Reply by BroJac5246** [reply]
Can you explain a bit about how extensions on iPads work? I'm not very familiar with the process. How hard should it be, and what was timing out? I have pretty much no experience with this but clearly you do :D

**Comment by WorldLanguages** [comment]
I was thinking, after the MV3 upgrade and some more testing, we could add an "Android" button to our website (scratchaddons.com) next to the Chrome, Firefox and Edge install buttons.
It would link to a doc explaining which Android browsers support installing browser extensions. We should probably encourage readers to use Firefox on Android, instead of Kiwi.
In that document, we could also 1) mention TurboWarp as an online alternative 2) clarify which type of devices we try to support (related: supporting touch screen, which also affects some laptops) and 3) a small additional note on iPad and iPhone support.

**Reply by BroJac5246** [reply]
That would be great! If we can, it would be good to let the install button at the top of the page also detect if you're on Android.

**Reply by Secret-chest** [reply]
But leave Kiwi as an alternative.

**Reply by WorldLanguages** [reply]
https://github.com/ScratchAddons/website-v2/issues/446

**Comment by Lukepuke311** [comment]
safari on mac seems very doable

**Reply by Samq64** [reply]
#7006
